styx-common/src/main/java/com/spotify/styx/util/TimeUtil.java

```diff
@@ -103,6 +103,34 @@ public static Instant nextInstant(Instant instant, Schedule schedule) {
         .toInstant();
   }
 
+  /**
+   * Gets the number of instants between lastInstant (inclusive) and firstInstant (inclusive)
+   * according to schedule.
+   *
+   * @param lastInstant The last instant
+   * @param firstInstant The first instant
+   * @param schedule The schedule of executions
+   * @return the number of instants in between
+   */
+  public static int numberOfInstants(Instant lastInstant, Instant firstInstant, Schedule schedule) {
+    if (lastInstant.isBefore(firstInstant)) {
+      throw new IllegalArgumentException();
+    }
+
+    final ExecutionTime executionTime = ExecutionTime.forCron(cron(schedule));
+
+    Instant currentInstant = lastInstant;
+    int number = 0;
+    while (!currentInstant.equals(firstInstant)) {
```

Comment:
A bit concerned about receiving instants that are not aligned with schedule (this might loop forever). Perhaps `currentInstant.isAfter(firstInstant)`?

Reply:
This is just a lib, the caller should take the responsibility in this case I think.

Reply:
I agree, but even if we push the alignment responsibility to the caller I think we should at least catch misalignment and throw an exception instead of getting stuck in an infinite loop.

Reply:
OK, if we really want to. To me it is over complicating things. I can make a fix.

Reply:
They are now enforced to be aligned https://github.com/spotify/styx/pull/402/files#diff-83fdfa5c37381e562bd0705651c031a6R115